packages/xo-server-sdn-controller/src/index.js

```diff
@@ -0,0 +1,789 @@
+
+    log.info(
+      `Private network '${
+        privateNetwork.name_label
+      }' has been created for pool '${pool.name_label}'`
+    )
+
+    // For each pool's host, create a tunnel to the private network
+    const hosts = filter(pool.$xapi.objects.all, { $type: 'host' })
+    await Promise.all(
+      map(hosts, async host => {
+        await this._createTunnel(host, privateNetwork)
+        this._createOvsdbClient(host)
+      })
+    )
+
+    const center = await this._electNewCenter(privateNetwork, false)
+    this._poolNetworks.push({
+      pool: pool.$ref,
+      network: privateNetwork.$ref,
+      starCenter: center ? center.$ref : null,
+    })
+    this._networks.set(privateNetwork.$id, privateNetwork.$ref)
+    if (center != null) {
+      this._starCenters.set(center.$id, center.$ref)
+    }
+  }
+
+  // ---------------------------------------------------------------------------
+
+  async _manageXapi(xapi) {
+    const { objects } = xapi
+
+    const objectsRemovedXapi = this._objectsRemoved.bind(this, xapi)
+    objects.on('add', this._objectsAdded)
+    objects.on('update', this._objectsUpdated)
+    objects.on('remove', objectsRemovedXapi)
+
+    await this._installCaCertificateIfNeeded(xapi)
+
+    return () => {
+      objects.removeListener('add', this._objectsAdded)
+      objects.removeListener('update', this._objectsUpdated)
+      objects.removeListener('remove', objectsRemovedXapi)
+    }
+  }
+
+  async _objectsAdded(objects) {
+    await Promise.all(
+      map(objects, async object => {
+        const { $type } = object
+
+        if ($type === 'host') {
+          log.debug(
+            `New host: '${object.name_label}' in pool: '${
+              object.$pool.name_label
+            }'`
+          )
+
+          if (!find(this._newHosts, { $ref: object.$ref })) {
+            this._newHosts.push(object)
+          }
+          this._createOvsdbClient(object)
+        }
+      })
+    )
+  }
+
+  async _objectsUpdated(objects) {
+    await Promise.all(
+      map(objects, async (object, id) => {
+        const { $type } = object
+
+        if ($type === 'PIF') {
+          await this._pifUpdated(object)
+        } else if ($type === 'host') {
+          await this._hostUpdated(object)
+        }
+      })
+    )
+  }
+
+  async _objectsRemoved(xapi, objects) {
+    await Promise.all(
+      map(objects, async (object, id) => {
+        const client = find(this._OvsdbClients, { id: id })
+        if (client) {
+          this._OvsdbClients.splice(this._OvsdbClients.indexOf(client), 1)
+        }
+
+        // If a Star center host is removed: re-elect a new center where needed
+        const starCenterRef = this._starCenters.get(id)
+        if (starCenterRef) {
+          this._starCenters.delete(id)
+          const poolNetworks = filter(this._poolNetworks, {
+            starCenter: starCenterRef,
+          })
+          for (let i = 0; i < poolNetworks.length; ++i) {
```

Comment:
Try to be consistent in your code.
Here you are using `++i` and `i++` here https://github.com/vatesfr/xen-orchestra/pull/4269/files#diff-d662a03f69531683c6900272a3b8e92aR415
In `for loop` there is no difference between `++i` and `i++`